src/popup/pages/AddToken/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import { Typography, TextField, Button, withStyles, WithStyles } from '@material-ui/core';
+import { inject, observer } from 'mobx-react';
+import cx from 'classnames';
+
+import styles from './styles';
+import NavBar from '../../components/NavBar';
+import AppStore from '../../stores/AppStore';
+import { handleEnterPress } from '../../../utils';
+
+interface IProps {
+  classes: Record<string, string>;
+  store: AppStore;
+}
+
+@inject('store')
+@observer
+class AddToken extends Component<WithStyles & IProps, {}> {
+  public componentDidMount() {
+    this.props.store.addTokenStore.init();
+  }
+
+  public render() {
+    const { classes, store: { addTokenStore } } = this.props;
+
+    return (
+      <div className={classes.root}>
+        <NavBar hasBackButton title="Add Token" />
+        <div className={classes.contentContainer}>
+          <div className={classes.fieldsContainer}>
+            <ContractAddressField onEnterPress={this.onEnterPress} {...this.props} />
+            {addTokenStore.name &&
+            (<div>
+              <DetailField fieldName={'Token Name'} value={addTokenStore.name || ''} classes={classes} />
```

Comment:
can handle the undefined value field in `DetailField` so you dont have to do this 3 times.